pkg/controller/collection/collection_controller.go

```diff
@@ -823,6 +829,15 @@ func reconcileActiveVersions(collectionResource *kabanerov1alpha1.Collection, co
 func getCollectionForSpecVersion(spec kabanerov1alpha1.CollectionVersion, collections []resolvedCollection) *resolvedCollection {
 	for _, collection := range collections {
 		if collection.collection.Version == spec.Version {
+			// Make sure all pipelines have a Sha256 set.  If they don't, complain but set a default one.
+			// Later on, if we attempt to retrieve the pipeline zip, it will fail.
+			for index, pipeline := range collection.collection.Pipelines {
+				if len(pipeline.Sha256) != 64 {
+					err := fmt.Errorf("Sha256 length %v is not valid", len(pipeline.Sha256))
+					log.Error(err, fmt.Sprintf("Collection ID %v version %v pipeline ID %v has an invalid Sha256 %v", collection.collection.Id, collection.collection.Version, pipeline.Id, pipeline.Sha256))
+					collection.collection.Pipelines[index].Sha256 = "0000000000000000000000000000000000000000000000000000000000000000"
```

Comment:
`strings.Repeat("0", 64)` might be more clear

Reply:
nice, I had no idea that existed.